sickrage/show/History.py

```diff
@@ -57,32 +57,55 @@ def get(self, limit=100, action=None):
         filter_sql = 'AND action in (' + ','.join(['?'] * len(actions)) + ') '
         order_sql = 'ORDER BY date DESC '
 
-        if limit == 0:
-            if actions:
-                results = self.db.select(common_sql + filter_sql + order_sql, actions)
-            else:
-                results = self.db.select(common_sql + order_sql)
+        if actions:
+            results = self.db.select(common_sql + filter_sql + order_sql, actions)
         else:
-            if actions:
-                results = self.db.select(common_sql + filter_sql + order_sql + 'LIMIT ?', actions + [limit])
-            else:
-                results = self.db.select(common_sql + order_sql + 'LIMIT ?', [limit])
+            results = self.db.select(common_sql + order_sql)
 
         data = []
-        for result in results:
-            data.append({
-                'action': result['action'],
-                'date': result['date'],
-                'episode': result['episode'],
-                'provider': result['provider'],
-                'quality': result['quality'],
-                'resource': result['resource'],
-                'season': result['season'],
-                'show_id': result['showid'],
-                'show_name': result['show_name']
-            })
-
-        return data
+        compact = []
+        data = [{'action': result['action'],
+                 'date': result['date'],
+                 'episode': result['episode'],
+                 'provider': result['provider'],
+                 'quality': result['quality'],
+                 'resource': result['resource'],
+                 'season': result['season'],
+                 'show_id': result['showid'],
+                 'show_name': result['show_name'],
+                } for result in results]
+
+        for row in data:
+            action = {
+                'action': row['action'],
+                'provider': row['provider'],
+                'resource': row['resource'],
+                'time': row['date']
+            }
+            if not any((history['show_id'] == row['show_id'] and
+                        history['season'] == row['season'] and
+                        history['episode'] == row['episode'] and
+                        history['quality'] == row['quality']) for history in compact):
+                history = {
+                    'actions': [action],
```

Comment:
For each row your moving [action] into the key 'actions'. You'l have to make gui changes of your going this route.